models/Document.php

```diff
@@ -674,16 +676,16 @@ public function setChildren($children)
      */
     public function getChildren($unpublished = false)
     {
-        if ($this->children === null) {
+        if (!isset($this->children[$unpublished])) {
```

Comment:
I'd go for a `getListingCacheKey()` method here as well, even though a bool get's converted to an int when used as array key, we should cast it in advance, just to avoid future issues :)